functions/v2/index.js

```diff
@@ -14,24 +14,21 @@
 
 'use strict';
 
+const functions = require('@google-cloud/functions-framework');
+
 // [START functions_cloudevent_pubsub]
-/**
- * CloudEvent function to be triggered by Pub/Sub.
- * This function is exported by index.js, and executed when
- * the trigger topic receives a message.
- *
- * @param {object} cloudevent A CloudEvent containing the Pub/Sub message.
- * @param {object} cloudevent.data.message The Pub/Sub message itself.
- */
-exports.helloPubSub = cloudevent => {
+// Register a CloudEvent callback with the Functions Framework that will
+// be executed when the Pub/Sub trigger topic receives a message.
+functions.cloudEvent('helloPubSub', cloudevent => {
```

Comment:
This will break our tests. How about configuring `functions.cloudEvent` to return the second argument, and _maybe_ (if it's possible) dropping the first?

Then we could do this:
```
exports.helloPubSub = functions.cloudEvent(cloudevent => {...})
```

Reply:
The API is already finalized and released so we can't change it now. 

Which tests will break? They all passed when I ran `npm run test`

Reply:
Oh, now I realize why: these tests use the `functions-framework` itself. What this *will* break are unit tests (which typically import the `export`ed function within Node itself).

Does `functions.cloudEvent` currently return anything? If not, how hard would it be to have it return the function itself?

Reply:
For additional context: the API in which the customer passes the function name to framework was a deliberate design decision. It removes the need for them to tell us the name of the function when they run the app (eg `functions-framework --target=helloPubSub`). This unblocks our ability to run multiple functions on the same port which is a [long standing feature request](https://github.com/GoogleCloudPlatform/functions-framework-nodejs/issues/23)

Reply:
right now `functions.cloudEvent` returns `void`: 

https://github.com/GoogleCloudPlatform/functions-framework-nodejs/blob/master/src/function_registry.ts#L70-L75

It makes sense to have it return something eventually, but I was trying to punt on that while we work out the design for some more features we are planning to add to the FFs. In my mind we will want to return the user's function wrapped in some object that exposes a configuration API:

```
functions
  .cloudEvent('foo', (ce) => ce.data)
  .afterSend(() => doSomeExpensiveCleanup())
```

but that is a bit speculative at the moment.

Reply:
Another possible pattern (though it feels less clean to me):

```
// Declare function
const helloPubSub = (cloudevent) => { ... }

// Expose function
functions.cloudEvent('helloPubSub', helloPubSub)
exports.helloPubSub = helloPubSub
```

Reply:
@ace-n I filed a bug in the functions framework to document this / explore API changes to make it easier: https://github.com/GoogleCloudPlatform/functions-framework-nodejs/issues/387

As it stands, this sample is an accurate representation of the existing API. Can we land it and work out the unit testing story in the FF repo?

Reply:
How do you feel about the pattern I suggested?

Reply:
I added some comments to the example above; if you can duplicate those in this PR that'd be 👍 .

Reply:
I am a bit confused, do you want me to implement this sample in that way? There are no unit tests for this sample so it seems unnecessary.

Reply:
Yes - that's my preference, for consistency reasons.